esp/services/ws_workunits/ws_workunitsHelpers.cpp

```diff
@@ -3004,6 +3005,23 @@ void WsWuHelpers::submitWsWorkunit(IEspContext& context, IConstWorkUnit* cw, con
     const char *paramXml, IArrayOf<IConstNamedValue> *variables, IArrayOf<IConstNamedValue> *debugs, IArrayOf<IConstApplicationValue> *applications)
 {
     ensureWsWorkunitAccess(context, *cw, SecAccess_Write);
+
+#ifndef _NO_LDAP
+    CLdapSecManager* secmgr = dynamic_cast<CLdapSecManager*>(context.querySecManager());
```

Comment:
I don't think security manager should queried and called if "checkViewPermissions" is not set in the config file.  A better way is for ws_workunits to check once at startup, set a member flag and wrap these calls in that test